bundles/org.eclipse.equinox.p2.publisher.eclipse/src/org/eclipse/equinox/internal/p2/publisher/eclipse/BrandingIron.java

```diff
@@ -504,7 +506,7 @@ private void copyMacIni(ExecutablesDescriptor descriptor, File initialRoot, File
 
 		StringBuffer buffer;
 		try {
-			buffer = readFile(ini);
+			buffer = new StringBuffer(Files.readString(ini.toPath(), StandardCharsets.UTF_8));
```

Comment:
I can't find in the original that UTF-8 is sued but system encoding instead.

Reply:
`In Java SE 9, properties files are loaded in UTF-8 encoding`
https://docs.oracle.com/javase/9/intl/internationalization-enhancements-jdk-9.htm#JSINT-GUID-974CF488-23E8-4963-A322-82006A7A14C7

Reply:
An (eclipse) ini file is not a java properties file...

Reply:
Yes, be really careful how this is written!  I know for sure that on Windows this is being written using the default encoding, not UTF-8 because I was just testing concerns that I have about Java 21 and its defaults.  I ran into cases where the installed IDE won't launch and then I look at they differences:

![image](https://github.com/eclipse-equinox/p2/assets/208716/c4c2ac12-b76b-47cf-91d7-756a23277e97)

Getting the encoding wrong is something you might not noticed until you installed into a folder with non-ASCII characters.  In the one case that works, the default system encoding is used and in the case that doesn't work, the debug launched installer I set to UTF-8 encoding.

I'm still not sure how this Java 21 stuff is going to work?  😱   Some of these files are read by the native launcher and I don't know what encoding it uses when reading these files nor what exactly it reads, but given the JRE is specified in the eclipse.ini, I matters what encoding is used...

Reply:
So we have at least 4 codes that read that file and interpret it differently?

org.eclipse.equinox.internal.frameworkadmin.equinox.EquinoxFwConfigFileParser.loadProperties(File)
org.eclipse.equinox.internal.frameworkadmin.equinox.utils.FileUtils.loadFile(File)
org.eclipse.equinox.internal.p2.publisher.eclipse.BrandingIron.copyMacIni(ExecutablesDescriptor, File, File, String)
https://github.com/eclipse-equinox/equinox/blob/8beb9ce0b1dea1176b06ad7300f2663ff6db4eff/features/org.eclipse.equinox.executable.feature/library/eclipseConfig.c#L146-L150

Reply:
i changed this commit to not touch the encoding

Reply:
This is what makes me so paranoid about suddenly things being UTF-8 on Windows where everything is currently some ISO-Latin type of encoding, at least on my OS.  But I know that other countries have different all kinds of other encodings ad their defaults...

Reply:
I'm not sure what you are suggesting.  In the past they are read and written using the default system encoding.   Both your changes change that to UTF-8 now.   If that's your intent, then you need to reconsider.  I think it should use Charset.defaultCharset() to preserve existing semantics.

Reply:
Hmm, I'm not sure when this branding iron thing runs.  Just at build time I suppose and then we are talking about the default encoding of the build-time process, so maybe it doesn't really matter to use UTF-8 always for this case...